clarity-ext-scripts/tests/integration/test_knm_sminet.py

```diff
@@ -0,0 +1,159 @@
+import os
+from lxml import etree as ET
+from uuid import uuid4
+import random
+import string
+import yaml
+import pytest
+from datetime import datetime
+from sminet_client import (SampleInfo, ReferringClinic, Patient, Doctor,
+                           SmiNetClient, SmiNetValidationError, SmiNetConfig, Laboratory)
+
+from clarity_ext_scripts.covid.sminet_settings import get_sminet_config, create_covid_request
+from clarity_ext_scripts.covid.knm_service import KNMConfig
+from clarity_ext_scripts.covid.knm_sminet_service import KNMSmiNetIntegrationService
+from clarity_ext_scripts.covid.partner_api_client import (
+    PartnerAPIV7Client, KARLSSON_AND_NOVAK, ORG_URI_BY_NAME)
+from clarity_ext.cli import load_config
+
+
+"""
+Tests the SmiNet/KNM integration
+"""
+
+try:
+    config = load_config()
+except ConfigNotFoundError:
+    pytest.xfail("This test requires a sminet_client configuration file")
+
+
+def generate_valid_contract_with_random_sample_id():
+    """
+    Returns a tuple of the parameters required to upload sample info via client.create 
+    """
+    constant_date = datetime(2020, 4, 30)
+    request_created = datetime(2020, 5, 18, 18, 11, 6)
+    timestamp = datetime.now().strftime("%y%m%dT%H%M%S")
+    prefix = "{}-".format(timestamp)
+    rnd = "".join(random.choice(string.ascii_uppercase + string.digits)
+                  for _ in range(25 - len(prefix)))
+    sample_id = prefix + rnd
+
+    sample_info = SampleInfo(status=1, sample_id=sample_id, sample_date_arrival=constant_date,
+                             sample_date_referral=constant_date, sample_material="Svalg",
+                             sample_free_text_referral="Extra info")
+    referring_clinic = ReferringClinic(
+        "Clinic name", "", "C", Doctor("Some doctor"))
+    patient = Patient("121212-1212", "k", "Tolvan Tolvansson", 23)
+    reporting_doctor = Doctor("Reporting Doctor")
+    laboratory = Laboratory(91, "National Pandemic Center at KI")
+    return create_covid_request(sample_info, referring_clinic, patient, reporting_doctor, laboratory)
+
+
+@pytest.mark.now()
+def test_can_create_request():
+    """
+    NOTE: We work against a whitelist on a jump server. Information available from team lead.
+    NOTE: Reads config from your clarity-ext config which should point to the test environment.
+          If sminet config is not available, the test is ignored. This is because the test
+          can't run on the build server because it's not whitelisted at the moment.
+    """
+
+    from clarity_ext_scripts.covid.knm_service import KNMSampleAccessor
+
+    integration = KNMSmiNetIntegrationService(config)
+    org_uri = ORG_URI_BY_NAME[KARLSSON_AND_NOVAK]
+    sample = KNMSampleAccessor(org_uri, "5236417647", datetime.now())
+    integration.export_to_sminet(sample)
+    return
+
+    #7224014063
+    print(organization)
+    print(patient)
+    print("INFO")
+    print(referring_clinic_name)
+
+    return
```

Comment:
Don't know how much work should be put into cleaning up this, since I guess it was mostly used as a dev-tool. But if you see further use for it, it might need some cleaning up. :smile:

Reply:
Yeah, all of this should go (the PR was perhaps a bit too much of a draft still :))

Reply:
This looks like some draft code, will remove it!

Reply:
Cleaned up! Was draft code